employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/rotation/ShiftTemplateEditForm.java

```diff
@@ -0,0 +1,242 @@
+import org.optaweb.employeerostering.shared.employee.Employee;
+import org.optaweb.employeerostering.shared.employee.EmployeeRestServiceBuilder;
+import org.optaweb.employeerostering.shared.rotation.RotationRestServiceBuilder;
+import org.optaweb.employeerostering.shared.rotation.ShiftTemplate;
+import org.optaweb.employeerostering.shared.rotation.view.ShiftTemplateView;
+import org.optaweb.employeerostering.shared.spot.Spot;
+import org.optaweb.employeerostering.shared.spot.SpotRestServiceBuilder;
+
+import static org.optaweb.employeerostering.gwtui.client.common.EventManager.Event.ROTATION_INVALIDATE;;
+
+@Templated
+public class ShiftTemplateEditForm extends AbstractFormPopup {
+
+    @DataField("from")
+    private RotationTimeSelector from;
+
+    @DataField("to")
+    private RotationTimeSelector to;
+
+    @DataField("spot")
+    private ListBox spotSelect; //FIXME: Don't use GWT widget
+
+    @DataField("employee")
+    private ListBox employeeSelect; //FIXME: Don't use GWT widget
+
+    @DataField("delete-button")
+    private HTMLButtonElement deleteButton;
+
+    @DataField("apply-button")
+    private HTMLButtonElement applyButton;
+
+    private TenantStore tenantStore;
+
+    private EventManager eventManager;
+
+    private TranslationService translationService;
+
+    private ShiftTemplateGridObject shiftTemplateGridObject;
+
+    private static final String SELECTED_CLASS = "selected";
+
+    @Inject
+    public ShiftTemplateEditForm(PopupFactory popupFactory, HTMLDivElement root, @Named("span") HTMLElement popupTitle, HTMLButtonElement closeButton,
+                                 HTMLButtonElement cancelButton, RotationTimeSelector from, RotationTimeSelector to,
+                                 ListBox spotSelect, ListBox employeeSelect,
+                                 HTMLButtonElement deleteButton, HTMLButtonElement applyButton,
+                                 TenantStore tenantStore, EventManager eventManager,
+                                 TranslationService translationService) {
+        super(popupFactory, root, popupTitle, closeButton, cancelButton);
+        this.from = from;
+        this.to = to;
+        this.spotSelect = spotSelect;
+        this.employeeSelect = employeeSelect;
+        this.deleteButton = deleteButton;
+        this.applyButton = applyButton;
+        this.tenantStore = tenantStore;
+        this.eventManager = eventManager;
+        this.translationService = translationService;
+    }
+
+    public void init(final ShiftTemplateGridObject shiftTemplateGridObject) {
+        this.shiftTemplateGridObject = (ShiftTemplateGridObject) shiftTemplateGridObject;
+        shiftTemplateGridObject.getElement().classList.add(SELECTED_CLASS);
+        final ShiftTemplateView template = shiftTemplateGridObject.getShiftTemplateModel().getShiftTemplateView();
+        setup(template);
+        showFor(shiftTemplateGridObject);
+    }
+
+    public void createNewShiftTemplate() {
```

Comment:
This method seems to be unused. Can you confirm?

Reply:
Confirmed; it a duplicate of the createNewShift() method. Might be useful to have later if we add a button to add a new Shift Template in the Rotation Screen, but as of now, unused.